### PR TITLE
C version added

### PR DESCRIPTION
**Pull request by gnbond**
Added a C version.  Optimized Part 2 version runs in about half a second.

I've used "git subtree" to include a unit-testing framework, I don't know how this will work out....

### CODE REVIEW

File: Part 1/c_solution/order_book.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "order_book.h"
+
+OrderBook *OrderBook_init(OrderBook *ob, const char *instrument) {
+    if (OrderList_init(&ob->buys) && OrderList_init(&ob->sells)) {
+        strlcpy(ob->instrument, instrument, INSTRUMENT_SIZE);
```
Comment (GaryHughes) [inline Part 1/c_solution/order_book.c]: strlcpy requires #include <bsd/string.h> on linux where the builds happen.

File: Part 1/c_solution/Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+tests:	$(TESTS)
+
+check: tests
+	@./test_order
+	@./test_orderlist
+	@./test_orderbook
+	@./test_booklist
+
+Exchange: $(OBJS)
+	$(CC) $(CFLAGS) -o $@ $(OBJS)
```
Comment (GaryHughes) [inline Part 1/c_solution/Makefile:25]: strlcpy requires -lbsd on linux where the builds happen

File: Part 2/c_solution/Makefile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	
+tests:	$(TESTS)
+
+check: tests
+	@./test_orderlist
+	@./test_orderbook
+	@./test_booklist
+
+Exchange: $(OBJS)
+	$(CC) $(CFLAGS) -o $@ $(OBJS)
```
Comment (GaryHughes) [inline Part 2/c_solution/Makefile:24]: strlcpy requires -lbsd on linux where the builds happen

File: Part 2/c_solution/order_book.c [hunk 4]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Implement the OrderBook.  Keep buys & sells for a single instrument.
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
```
Comment (GaryHughes) [inline Part 2/c_solution/order_book.c:9]: strlcpy requires #include <bsd/string.h> on linux where the builds happen

**Review by GaryHughes — CHANGES_REQUESTED** [review]
Some small changes required to compile on linux.

**Comment by gnbond** [comment]
Committed a change to use strncpy() instead.  Builds on Ubuntu.

**Pull request merged**